openapi/openapi_client_test.go

```diff
@@ -1029,4 +1421,75 @@ func TestProviderClientDelete(t *testing.T) {
 			})
 		})
 	})
+
+	Convey("Given a providerClient set up with stub client that returns some response", t, func() {
+		httpClient := &http_goclient.HttpClientStub{}
+		providerClient := &ProviderClient{
+			openAPIBackendConfiguration: &specStubBackendConfiguration{
+				host:        "wwww.host.com",
+				basePath:    "/api",
+				httpSchemes: []string{"http"},
+			},
+			httpClient:            httpClient,
+			providerConfiguration: providerConfiguration{},
+			apiAuthenticator: &specStubAuthenticator{
+				authContext: &authContext{
+					headers: map[string]string{},
+				},
+			},
+		}
+		Convey("When providerClient DELETE  method is called with a SpecV2Resource that has a subresource path, a requestPayload, an empty responsePayload and the resource parentID", func() {
+			specv2Resource := &SpecV2Resource{
+				Path: "/v1/resource/{id}/subresource",
+				RootPathItem: spec.PathItem{
+					PathItemProps: spec.PathItemProps{
+						Post: &spec.Operation{
+							VendorExtensible: spec.VendorExtensible{},
+							OperationProps: spec.OperationProps{
+								Responses: &spec.Responses{
+									ResponsesProps: spec.ResponsesProps{
+										StatusCodeResponses: map[int]spec.Response{
+											201: {},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				InstancePathItem: spec.PathItem{
+					PathItemProps: spec.PathItemProps{
+						Delete: &spec.Operation{
+							VendorExtensible: spec.VendorExtensible{},
+							OperationProps: spec.OperationProps{
+								Responses: &spec.Responses{
+									ResponsesProps: spec.ResponsesProps{
+										StatusCodeResponses: map[int]spec.Response{
+											204: {},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			parentIDs := []string{"parentID"}
+			expectedID := "1234"
+			_, err := providerClient.Delete(specv2Resource, expectedID, parentIDs...)
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("And then client should have received the right URL", func() {
+				expectedProtocol := providerClient.openAPIBackendConfiguration.getHTTPSchemes()[0]
+				expectedHost, _ := providerClient.openAPIBackendConfiguration.getHost()
+				expectedBasePath := providerClient.openAPIBackendConfiguration.getBasePath()
+				So(httpClient.URL, ShouldEqual, fmt.Sprintf("%s://%s%s/v1/resource/%s/subresource/%s", expectedProtocol, expectedHost, expectedBasePath, parentIDs[0], expectedID))
```

Comment:
I feel that `So(httpClient.URL, ShouldEqual, "http://wwww.host.com/api/v1/resource/parentID/subresource/1234")` would make this test far more valuable in terms of readable documentation.

Reply:
+1